ivy/functional/backends/torch/extensions.py

```diff
@@ -121,3 +121,24 @@ def lcm(
 
 
 lcm.support_native_out = True
+
+
+def hann_window(
+    window_length: int,
+    periodic: Optional[bool] = True,
+    dtype: Optional[torch.dtype] = None,
+    *,
+    out: Optional[torch.tensor] = None,
+) -> torch.tensor:
+    return torch.hann_window(
+        window_length, 
+        periodic=periodic, 
+        dtype=dtype, 
+        layout=torch.strided,
+        device=None,
+        requires_grad=None
+    )
+
+
+hann_window.support_native_out = False
+hann_window.unsupported_dtypes = ("int32", "int64")
```

Comment:
It seems this part about unsupported dtypes is not required. Tests pass for me. Not sure if you had something on mind here.